nhd/NHDScheduler.py

```diff
@@ -473,9 +491,23 @@ def run(self):
                     if idle_cnt >= IDLE_CNT_THRESH:
                         idle_cnt = 0
                         self.CheckPendingPods()
+                        
+                        if new_nodes_count > 0:
```

Comment:
Why is this outer condition needed? Shouldn't this work be done whenever the `NHDWatchTypes.NHD_WATCH_TYPE_TRIAD_NODE_CREATE` event is received?

Reply:
The triad node create event happens whenever a new node shows up in the cluster but is not necessarily in "Ready" state yet. It spends several seconds in NotReady state initially. The exact time it spends transferring from NotReady to Ready varies. It can take anywhere between a few seconds to a few tens of seconds.
NHD scheduler can only scan and detect nodes in Ready state.

To deal with this I increment the variable new_nodes_count in response to the NHD_WATCH_TYPE_TRIAD_NODE_CREATE event:
https://github.com/psiwczak/nhd/blob/f298f7d055871da2d7ce4c41006111388b1d5e13/nhd/NHDScheduler.py#L509

this variable is used to trigger the node scan  in the main scheduler loop which will catch all new nodes in the Ready state. The scan will be triggered only if new_nodes_count > 0.